crates/graphql_artifact_generation/src/reader_ast.rs

```diff
@@ -125,21 +125,23 @@ fn linked_field_ast_node(
 
     let condition = match linked_field.associated_data.variant {
         SchemaServerFieldVariant::InlineFragment(_) => {
-            let object = schema
-                .server_field_data
-                .object(linked_field.associated_data.parent_object_id);
+            match linked_field.associated_data.parent_object_id {
+                LinkedType::LinkedServerObject(parent_object_id) => {
```

Comment:
Hm. We shouldn't have an inline variant and a linked client field. Happy to panic when necessary, for now, but if convenient, maybe we put the variant inside of the `TServerObject`? e.g.
```
pub struct ValidatedLinkedFieldAssociatedData {
    pub parent_object_id: LinkedType<StructWithServerObjectIdAndVariant>,
```

thoughts?

note that in reader_ast, we check the variant inside of the `LinkedServerObject` case, so that would work there, too

Reply:
Since `SchemaServerFieldInlineFragmentVariant` has `pub server_field_id: ServerFieldId` already
I've removed `LinkedType` variant and added `field_id` to `SchemaServerFieldLinkedFieldVariant`
```rust
#[derive(Debug, Clone)]
pub enum SchemaServerFieldVariant {
    InlineFragment(SchemaServerFieldInlineFragmentVariant),
    LinkedField(SchemaServerFieldLinkedFieldVariant),
}

#[derive(Debug, Clone)]
pub struct SchemaServerFieldLinkedFieldVariant {
    pub field_id: FieldType<ServerFieldId, ()>,
}
```
This way in both cases we have access to the `field` which has access to `parent_object_id`